synapse/rest/client/v2_alpha/register.py

```diff
@@ -379,7 +383,15 @@ def on_POST(self, request):
         if self.hs.config.user_consent_at_registration:
             new_flows = []
             for flow in flows:
-                flow.append(LoginType.TERMS)
+                inserted = False
+                # m.login.terms should go near the end but before msisdn or email auth
+                for i, stage in enumerate(flow):
+                    if stage == LoginType.EMAIL_IDENTITY or stage == LoginType.MSISDN:
+                        flow.insert(i, LoginType.TERMS)
+                        inserted = True
+                        break
+                if not inserted:
```

Comment:
ftr you can do:

```
for ... :
   # ...
else:
   # we only get here if we fell off the end of the for
```

I'm unsure if the gain in not having to mess with `inserted` outweighs the loss of using a bit of an obscure language feature. up to you.

Reply:
Oh yes - I guess whichever is more in-keeping with the code style, but personally I had to stop & think what that syntax does both times I've encountered it so I'd be inclined to keep it as-is.